### PR TITLE
feat: Add SUB_SPECIALTY curricula to validator

### PR DESCRIPTION
**Pull request by Marcellofabbri**
Validator now checks curricula are of curriculumSubType "SUB_SPECIALTY" as well as "MEDICAL_CURRICULUM"

**TIS21-1674**: add list of subspecialties to the Form R Part A Declarations Section

### CODE REVIEW

File: src/test/java/uk/nhs/hee/tis/trainee/reference/dto/validator/CurriculumValidatorTest.java [hunk 1]
```diff
@@ -49,6 +49,7 @@ void setUp() {
       "DELETE,MEDICAL_CURRICULUM,false",
       "CURRENT,Other,false",
       "CURRENT,MEDICAL_CURRICULUM,true",
+      "CURRENT,SUB_SPECIALTY, true",
```
Comment (JosephKelly) [inline src/test/java/uk/nhs/hee/tis/trainee/reference/dto/validator/CurriculumValidatorTest.java:52]: For the sake of completeness (branch coverage), we should add the other test cases `"INACTIVE,SUB_SPECIALTY,false",...`.

**Review by JosephKelly — CHANGES_REQUESTED** [review]
There's a change that probably merits bumping the version number up too

**Comment by Marcellofabbri** [comment]
Right, I've experimented a bit and found out what should happen if we retrigger the synchronization of the Curriculum table with the existing changes in tis-trainee-reference.

It looks like any new Curriculum being saved in the reference collection will indeed have the `curriculumSubType` attribute from now on. But as it is, it won't update the existing ones with their `curriculumSubType`, as those documents get updated only in the fields that weren't null (i.e. `tisid`, `label`), because the mapper ignores fields that are null in the record that's getting updated. So basically it would work for anything new but wouldn't update anything old, cos all the `"MEDICAL_CURRICULUM"` have no `curriculumSubtype`, while the new ones would be inserted correctly with `curriculumSubType` declared as `"SUB_SPECIALITY"`.

To allow for existing curricula (the `"MEDICAL_CURRICULUM"` ones) to be updated with the `curriculumSybType` value the CurriculumMapper needs to be changed with this line so that it will overwrite null fields instead of ignoring them: 
**@Mapping(target = "curriculumSubType"~~, ignore = true~~)**

File: src/main/java/uk/nhs/hee/tis/trainee/reference/mapper/CurriculumMapper.java [hunk 2]
```diff
@@ -40,5 +40,6 @@
 
   @Mapping(target = "id", ignore = true)
   @Mapping(target = "tisId", ignore = true)
+  @Mapping(target = "curriculumSubType")
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/mapper/CurriculumMapper.java:43]: This line is not needed, mapping is automatic for matching fields.

**Review by Judge40 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/nhs/hee/tis/trainee/reference/mapper/CurriculumMapper.java (hunk 2 above)
Comment (Judge40) [reply src/main/java/uk/nhs/hee/tis/trainee/reference/mapper/CurriculumMapper.java:43]: This is the generated code without the manual mapping.

```java
@Override
public Curriculum update(Curriculum target, Curriculum source) {
    if ( source == null ) {
        return null;
    }
    target.setLabel( source.getLabel() );
    target.setCurriculumSubType( source.getCurriculumSubType() );
    return target;
}
```

**Review by Judge40 — APPROVED** [review]

**Pull request merged**